### PR TITLE
Allow set_montage to take Digitization

### PR DESCRIPTION
**Pull request by massich**

### CODE REVIEW

File: mne/channels/tests/test_channels.py [hunk 1]
```diff
@@ -312,7 +312,7 @@ def test_set_montage():
     from mne.io import RawArray
     from mne.utils import object_diff
 
-    ch_names = ['EEG {id:03d}'.format(id=id) for id in range(1, 61)]
+    ch_names = ['EEG {id:03d}'.format(id=id) for id in range(61)]
     raw = RawArray(data=np.empty([len(ch_names), 1]),
                    info=create_info(ch_names=ch_names, sfreq=1))
     original_raw = (read_raw_fif(raw_fname, preload=False)
```
Comment (massich) [inline mne/channels/tests/test_channels.py]: when reading `chs` contains `EEG 001` - `EEG 060`, but dig contains from 0 to 60. there's no notion of name, but the first one is the one that gets dropped

Comment (larsoner) [reply mne/channels/tests/test_channels.py]: Is this neuromag? The first digitized point is the reference electrode

Comment (larsoner) [reply mne/channels/tests/test_channels.py]: (or maybe it's the last? Can't remember, but you can check by comparing to `info['chs'][eeg_idx]['loc']`)

Comment (drammock) [reply mne/channels/tests/test_channels.py]: I think ref was first (after cardinals) in our (neuromag) lab.

**Comment by agramfort** [comment]
should be ok now.

thanks the hint guys !

>

File: mne/channels/channels.py [hunk 2]
```diff
@@ -464,8 +466,34 @@ def set_montage(self, montage, set_dig=True, verbose=None):
...
+
+            for ch, digpoint in zip(self.info['chs'], eeg_montage):
+                ch['kind'] = _human2fiff['eeg']
+                ch['unit'] = FIFF.FIFF_UNIT_V
+                ch['coil_type'] = FIFF.FIFFV_COIL_EEG
+                ch['coord_frame'] = digpoint['coord_frame']
+                ch['loc'] = np.zeros(12)
+                ch['loc'][:3] = digpoint['r']
+                ch['loc'][3:6] = ref_pos
+                ch['loc'][[7, 11]] = 1.  # not clear why
```
Comment (teonbrooks) [inline mne/channels/channels.py]: i know for meg coils, idx 3-11 are the basis for neuromag space, i believe: https://github.com/mne-tools/mne-python/blob/master/mne/io/kit/kit.py#L704

**Comment by agramfort** [comment]
This one can be closed now

**Pull request closed without merging**